lhotse/dataset/speech_synthesis.py

```diff
@@ -65,15 +69,21 @@ def __getitem__(self, cuts: CutSet) -> Dict[str, torch.Tensor]:
             features = transform(features)
 
         tokens, tokens_lens = self.token_collater(cuts)
-
-        return {
+        batch = {
             "audio": audio,
             "features": features,
             "tokens": tokens,
             "audio_lens": audio_lens,
             "features_lens": features_lens,
             "tokens_lens": tokens_lens,
         }
+        if self.speaker_id_mapping is not None:
+            batch["speakers"] = torch.tensor(
+                [self.speaker_id_mapping[cut.supervisions[0].speaker] for cut in cuts],
+                dtype=torch.long,
+            )
+        else:
```

Comment:
This `else:` is a bug, you'd skip `return batch` when `speaker_id_mapping` is available.